src/utils.ts

```diff
@@ -1,6 +1,8 @@
 import { TabsMapUpdater } from "./types";
 
 export function ensureClosingSlash(url: string): string {
+  if (!url.trim()) throw new Error("url input cannot be empty");
+
```

Comment:
I would prefer the error message to be: "Invalid input: " + url.

Also please extend the condition check with: `typeof url !== "string"`.